recipe/meta.yaml

```diff
@@ -12,50 +12,72 @@ source:
 build:
   number: 0
   skip: true  # [py<37 or not unix]
-  script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv
 
 requirements:
   host:
     - pip
     - python
   run:
-    - aiofiles >=0.7.0,<0.8.0
-    - ansimarkup >=1.0.0
-    - colorama >=0.4,<1.0
-    - graphene >=2.1,<3
-    - jinja2 ==2.11.0,<2.12
-    - metomi-isodatetime >=1!2.0.2, <1!2.1.0
-    - protobuf >=3.15.0,<3.16.0
-    - psutil >=5.6.0
     - python
-    - pyuv >=1.4.0,<1.5.0
-    - pyzmq >=19.0.0,<19.1.0
-    - setuptools >=49
-    - urwid >=2,<3
-    # TODO: optional dependencies, see discussion here https://github.com/conda-forge/cylc-flow-feedstock/pull/4
-    - empy >=3.3,<3.4
-    - pandas >=1.0
-    - pympler
-    - matplotlib-base
 
 test:
   imports:
     - cylc.flow
   commands:
     - cylc --version
+    - cylc version --long
+    - cylc extract-resources etc/job.sh  # check packaging
+
+outputs:
+  # base package with only required dependencies
+  - name: {{ name }}-base
+    build:
+      script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv
+    requirements:
+      host:
+        - pip
+        - python
+      run:
+        - aiofiles >=0.7.0,<0.8.0
+        - ansimarkup >=1.0.0
+        - colorama >=0.4,<1.0
+        - graphene >=2.1,<3
+        - jinja2 ==2.11.0,<2.12
+        - metomi-isodatetime >=1!2.0.2, <1!2.1.0
+        - protobuf >=3.15.0,<3.16.0
+        - psutil >=5.6.0
+        - pyuv >=1.4.0,<1.5.0
+        - pyzmq >=19.0.0,<19.1.0
+        - setuptools >=49
+        - urwid >=2,<3
+
+  # full package with all dependencies installed
+  - name: {{ name }}
+    build:
+      script: {{ PYTHON }} -m pip install .[all] --no-deps --ignore-installed -vv
```

Comment:
Should replace `[all]` with a more specific list to exclude dev deps.

Should add a new opt-dep to our packages to make this easier in future.